Source/Basetypes/BigFloat.cs

```diff
@@ -157,6 +157,56 @@ public static BigFloat FromBigDec(BigDec v, int significandSize, int exponentSiz
       return new BigFloat(isNeg, sig, exp, sigSize, expSize);
     }
 
+    [Pure]
+    public static BigFloat FromBVString(String s)
+    {
+      /*
+       * String must be either of the format (fp (bv* *) (bv* *) (bv* *))
+       * or of the special value format: (_ val * *)
+       * Where * indicates an integer value
+       * and val is one of the special values: NaN, +oo, -oo, zero, -zero
+       */
+      BIM sig, exp;
```

Comment:
Declare variables where they are first used. No need to declare them in the beginning of a function.